nixos/modules/services/networking/tailscale.nix

```diff
@@ -115,7 +115,9 @@ in {
       script = ''
         status=$(${config.systemd.package}/bin/systemctl show -P StatusText tailscaled.service)
         if [[ $status != Connected* ]]; then
```

Comment:
What is the usecase though? This will only work if tailscale is not connected yet, but in this case it won't work without auth-key

Reply:
I used it to set option like this.
```
services.tailscale.enable = true;
services.tailscale.useRoutingFeatures = "both";
services.tailscale.extraUpFlags = [ "--ssh" "--advertise-exit-node" ];
```

I think you are correct. I misunderstood the action. Didn't know there was set command.

Reply:
Created new option `extraSetFlags`.

Reply:
I still don' 100% see the usecase.
If we think about a scenario where we deploy a new server with our NixOS configuration and use `extraSetFlags` we would expect that once the server is deployed we only need to SSH into it run `tailscale up`, login and authorize the new machine and after that the server should e.g., advertise itself as an exit node.

This, at least in my tests, didn't work. The changes made by `tailscale set` are simply reverted when doing the first `tailscale up`. After doing `tailscale up` we could obviously do `systemctl restart tailscaled-set.service` and it would work again, but I don't think this is the desired behavior.

It would be great if someone could also try to reproduce this. I used `tailscale debug prefs` to print the currently used settings.

Reply:
I wrote this PR because I was not using "authkey" option and there was no way to configure daemon in nix. #276912 
Rebuild / Reboot or restarting oneshot service as you mentioned makes this option work. 
I admit that this option is tricky.. but couldn't find better solution..

```
services.tailscale.enable = true;
services.tailscale.useRoutingFeatures = "both";
services.tailscale.extraSetFlags = ["--ssh" "--advertise-exit-node=true"];
```

Reply:
Ok, then we should definitely document that a restart of the oneshot service is required **after** running `tailscale up`.
Because currently it's not obvious when/how this option works and it's sadly also not at all declarative 😔